### PR TITLE
Update consul chart

### PR DESCRIPTION
**Pull request by apenney**
Replaces #361 

I've been working on updating @lachie83 's PR, and was going to see if he'd pull my changes in.  It may be easier to just handle this in a new PR.

I've been playing with this for a few days and it seems to work and is an incremental improvement over the existing consul chart.  I'm hoping we can merge this in then seek for future improvements with time.

### CODE REVIEW

**Comment by lachie83** [comment]
@apenney This is awesome. Would love to pull this in and make this stable. I'm busy for the next week or so but will work on it as soon as things free up on my side. Thanks again!

**Comment by lachie83** [comment]
@k8s-bot ok to test

File: stable/consul/templates/consul.yaml [hunk 1]
```diff
@@ -152,18 +181,16 @@ spec:
       volumes:
       - name: gossip-key
         secret:
-          secretName: gossip-key
-      readinessProbe:
```
Comment (lachie83) [inline stable/consul/templates/consul.yaml:156]: @apenney any ideas on how we can implement a health check?

**Comment by lachie83** [comment]
I would like to see a healthcheck on this chart. Any ideas @apenney ?

**Comment by lachie83** [comment]
Happy to push this out and figure out the health checks over time.

File: stable/consul/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name: consul
 home: https://github.com/hashicorp/consul
-version: 0.1.4
-description: Highly available and distributed service discovery and key-value store designed with support for the modern data center to make distributed systems and configuration easy.
+version: 0.2.0
+description: Hashicorp Consul Helm chart for Kubernetes.
```
Comment (prydonius) [inline stable/consul/Chart.yaml]: Can we keep old description?

File: stable/consul/README.md [hunk 3]
```diff
@@ -48,7 +45,9 @@ The following tables lists the configurable parameters of the consul chart and t
 | `SerfwanUdpPort`        | Container serf wan UDP listening port | `8302`                                                     |
 | `ServerPort`            | Container server listening port       | `8300`                                                     |
 | `ConsulDnsPort`         | Container dns listening port          | `8600`                                                     |
-
+| `ui.enabled`            | Enable Consul Web UI                  | `false`                                                    |
+| `uiSerivce.enable`      | Create dedicated Consul Web UI svc    | `false`                                                    |
```
Comment (prydonius) [inline stable/consul/README.md]: uiService

File: stable/consul/templates/consul.yaml [hunk 4]
```diff
@@ -1,15 +1,34 @@
 apiVersion: v1
 kind: Secret
 metadata:
-  name: gossip-key
+  name: {{ .Release.Name }}-gossip-key
```
Comment (prydonius) [inline stable/consul/templates/consul.yaml]: should this be `{{ template "fullname" . }}-gossip-key`?

File: stable/consul/README.md (hunk 3 above)
Comment (prydonius) [inline stable/consul/README.md]: enabled

**Comment by lachie83** [comment]
@prydonius thanks for the review. I've addressed your concerns. Please take another pass.

**Comment by seanknox** [comment]
Reviewed and tested with @lachie83. 👍

**Pull request merged**